examples/android-room/src/main/java/io/sentry/android/roomsample/ui/MainActivity.kt

```diff
@@ -0,0 +1,86 @@
+package io.sentry.android.roomsample.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.util.AttributeSet
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.activity.ComponentActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.sentry.Sentry
+import io.sentry.android.roomsample.R
+import io.sentry.android.roomsample.SampleApp
+import io.sentry.android.roomsample.data.Track
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlin.concurrent.timerTask
+import kotlin.system.measureTimeMillis
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val list = findViewById<RecyclerView>(R.id.list)
+        list.layoutManager = LinearLayoutManager(this)
+        list.adapter = TrackAdapter()
+
+        lifecycleScope.launchWhenStarted {
+            val tracks: List<Track> =
+                SampleApp.database.tracksDao().allByArtist("Red Hot Chili Peppers")
+            (list.adapter as TrackAdapter).populate(tracks)
+        }
+
+        lifecycleScope.launchWhenResumed {
+            // this is to make sure that DB query spans are sent to Sentry, as they are attached to the Cold Start Transaction, otherwise they get dropped
+            delay(1000)
+            Sentry.getSpan()?.finish()
```

Comment:
should we call this after `(list.adapter as TrackAdapter).populate(tracks)` then since the DB did its job already?

Reply:
True, my brain was damaged by bytecode, hence i did some shit here 🤣